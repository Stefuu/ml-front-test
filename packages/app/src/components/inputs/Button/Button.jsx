import styled from 'styled-components'
import { theme } from 'styled-tools'
import { height, space, width } from 'styled-system'
import tag from 'clean-tag'

const Button = styled(tag.button)`
    cursor: pointer;
    outline: none;
    border-radius: 6px;
    color: ${theme('colors.b0')};
    background-color: ${theme('colors.primary')};
    ${height}
    ${space}
    ${width}
`

export default Button
