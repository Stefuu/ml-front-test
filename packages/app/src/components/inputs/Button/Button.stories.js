import React from 'react'
import { storiesOf } from '@storybook/react'
import { doc } from 'storybook-readme'
import readmeFile from './README.md'
import centered from '@storybook/addon-centered'
import { withKnobs, color, number } from '@storybook/addon-knobs'
import theme from '../../themes/default'
import { ThemeProvider } from 'styled-components'
import Button from './Button'
import GlobalStyle from '../../modules/GlobalStyle'

storiesOf('app/components/elements/Button', module)
  .addParameters({ jest: ['Button.test.js'] })
  .add('README', doc(readmeFile))
  .addDecorator(centered)
  .addDecorator(withKnobs)
  .addParameters({ jest: ['Content.test.js'] })
  .add('simple', () => (
    <ThemeProvider theme={theme}>
      <React.Fragment>
        <GlobalStyle />
        <Button color={color('color', '')} width={number('width', 170)} height={number('height', 56)}>Test</Button>
      </React.Fragment>
    </ThemeProvider>
  ))
