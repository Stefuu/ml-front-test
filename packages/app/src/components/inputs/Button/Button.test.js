import React from 'react'
import { render } from 'react-testing-library'
import Button from './Button'

describe('Button', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
