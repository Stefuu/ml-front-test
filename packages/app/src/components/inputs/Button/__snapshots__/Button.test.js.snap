// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render [snapshot] 1`] = `
<button
  class="sc-bdVaJa cGdrmk"
/>
`;
